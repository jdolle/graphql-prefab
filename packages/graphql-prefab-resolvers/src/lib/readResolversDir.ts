import path from 'path'
import { readdirSync } from 'fs'
import { filter, map } from 'ramda'

const isResolverFile = (filename: string) => filename.endsWith('.json')

const filterResolvers = filter(isResolverFile)

const readResolversDir = (dir: string) => {
  const files = readdirSync(dir)
  const resolverFilters = filterResolvers(files)

  console.assert(resolverFilters.length, 'No resolver configs found.')
  const concatenatePathnames = map((fileName: string) => path.join(dir, fileName))

  return concatenatePathnames(resolverFilters)
}

export default readResolversDir
