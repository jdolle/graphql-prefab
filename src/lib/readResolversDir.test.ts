import path from 'path'
import readResolversDir from './readResolversDir'

it('#readResolversDir gets all json files in the directory', () => {
  const dirPath = path.resolve(__dirname, '../../examples/demo-server/resolvers')
  expect(readResolversDir(dirPath)).toMatchSnapshot()
})

it('#readResolversDir throws on an invalid path', () => {
  expect(() => readResolversDir('./does_not_exist')).toThrow()
})
