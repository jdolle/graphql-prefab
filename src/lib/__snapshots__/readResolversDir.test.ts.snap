// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#readResolversDir gets all json files in the directory 1`] = `
Array [
  "Query.json",
]
`;
